pkg/proto/v0/accounts.pb.micro_test.go

```diff
@@ -0,0 +1,944 @@
+	assertResponseContainsGroup(t, resp, updatedGroup)
+
+	cleanUp(t)
+}
+
+// https://github.com/owncloud/ocis-accounts/issues/62
+func TestAddMemberAlreadyInGroup(t *testing.T) {
+	grp1 := getTestGroups("grp1")
+	account := getAccount("user1")
+
+	updatedGroup := grp1
+	updatedGroup.Members = append(updatedGroup.Members, &proto.Account{Id: account.Id})
+
+	createGroup(t, grp1)
+	createAccount(t, account.PreferredName)
+
+	addMemberToGroup(t, grp1.Id, account.Id)
+
+	client := service.Client()
+	cl := proto.NewGroupsService("com.owncloud.api.accounts", client)
+
+	req := &proto.AddMemberRequest{GroupId: grp1.Id, AccountId: account.Id}
+
+	res, err := cl.AddMember(context.Background(), req)
+
+	// Should Give Error
+	checkError(t, err)
+	assert.IsType(t, &proto.Group{}, res)
+	//assert.Equal(t, proto.Group{}, *res)
+	//assertGroupsSame(t, updatedGroup, res)
+
+	// Check the group is truly updated
+	resp := listGroups(t)
+	assertResponseContainsGroup(t, resp, updatedGroup)
+
+	cleanUp(t)
+}
+
+func TestAddMemberNonExisting(t *testing.T) {
+	grp1 := getTestGroups("grp1")
+
+	createGroup(t, grp1)
+
+	client := service.Client()
+	cl := proto.NewGroupsService("com.owncloud.api.accounts", client)
+
+	invalidIds := []string{
+		"$@dsfd",
+		"42",
+		"happyString",
+		"0ed84f08-aa0a-46e4-8e42-f0a5d6e1b059",
+		"  ",
+	}
+
+	for _, id := range invalidIds {
+		req := &proto.AddMemberRequest{GroupId: grp1.Id, AccountId: id}
+
+		res, err := cl.AddMember(context.Background(), req)
+		assert.IsType(t, &proto.Group{}, res)
+		assert.Empty(t, res)
+		assert.Error(t, err)
+		assert.Equal(
+			t,
+			fmt.Sprintf("{\"id\":\".\",\"code\":404,\"detail\":\"could not read account: open accounts-store/accounts/%v: no such file or directory\",\"status\":\"Not Found\"}", id),
+			err.Error(),
+		)
+	}
+
+	// Check group is not changed
+	resp := listGroups(t)
+	assertResponseContainsGroup(t, resp, grp1)
+
+	cleanUp(t)
+}
+
+func addMemberToGroup(t *testing.T, groupId, memberId string) (*proto.Group, error) {
+	client := service.Client()
+	cl := proto.NewGroupsService("com.owncloud.api.accounts", client)
+
+	req := &proto.AddMemberRequest{GroupId: groupId, AccountId: memberId}
+
+	res, err := cl.AddMember(context.Background(), req)
+
+	return res, err
+}
+
+// https://github.com/owncloud/ocis-accounts/issues/61
+func TestRemoveMember(t *testing.T) {
+	grp1 := getTestGroups("grp1")
+	account := getAccount("user1")
+
+	createGroup(t, grp1)
+	createAccount(t, account.PreferredName)
+
+	addMemberToGroup(t, grp1.Id, account.Id)
+
+	client := service.Client()
+	cl := proto.NewGroupsService("com.owncloud.api.accounts", client)
+
+	req := &proto.RemoveMemberRequest{GroupId: grp1.Id, AccountId: account.Id}
+
+	res, err := cl.RemoveMember(context.Background(), req)
+	checkError(t, err)
+
+	assert.IsType(t, &proto.Group{}, res)
+	//assert.Equal(t, proto.Group{}, *res)
+	// assertGroupsSame(t, grp1, res)
+
+	resp := listGroups(t)
+	assertResponseContainsGroup(t, resp, grp1)
```

Comment:
need also to check that the member is not in the group anymore

Reply:
The original grp1 doesn't contain the member, in the beginning, if the response contains grp1, then we can assume that the member was removed from the group.